### DISCUSSION TITLE
The calculation result of case  Onera M6 wing is different from the tutorial result

### DISCUSSION THREAD

**Discussion by hxl1310** [mdolab/dafoam#471] [Tutorials]
I follow the tutorial

./preProcessing.sh
mpirun -np 8 python runScript.py 2>&1 | tee logOpt.txt

The final calculation result is very different from the tutorial.

The tutorial initial drag is 0.016597241 and the optimized drag is 0.013540159 with a drag reduction of 18.4%.
but my calculation result

![微信图片_20230820183009](https://github.com/mdolab/dafoam/assets/118341727/23460a12-3bb9-4932-a7c6-bf69223d7cdc)

**Comment by friedenhe** [comment]
The numbers from the tutorial website were for the refined mesh, while the default mesh density in the tutorial files is very coarse. Please try to refine the mesh and see if you can reproduce the numbers reported on the tutorial website.

**Reply by jCaudron** [reply]
Dear Sir,

I am encountering the same issue. I attempted to refine the mesh by removing the "coarsen" instruction from the pre-processing file, yet I am still unable to match the provided results. Could you please share the mesh parameters that were used to achieve those results? Specifically, I am interested in the values of N, s0, and marchDist, or any other parameters that need modification.

Thank you!

**Comment by hxl1310** [comment]
I am also unable to obtain the results shown in the tutorial. Remove the bold command and refine the mesh to only approximate the result.


···生活···
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mdolab/dafoam"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年3月5日(星期二) 晚上6:06
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mdolab/dafoam] The calculation result of case  Onera M6 wing is different from the tutorial result (Discussion #471)





 
Dear Sir,
 
I am encountering the same issue. I attempted to refine the mesh by removing the "coarsen" instruction from the pre-processing file, yet I am still unable to match the provided results. Could you please share the mesh parameters that were used to achieve those results? Specifically, I am interested in the values of N, s0, and marchDist, or any other parameters that need modification.
 
Thank you!

**Comment by friedenhe** [comment]
We don't remember the exact N, s0 we used for the M6 tutorial figures. But I suggest you try to use N=60 to 80 and choose s0 such that the y+ is about 30.

**Reply by Mohsin-Bashir** [reply]
I have tried these parameters but still could not reproduce the numbers mentioned on tutorial website. 
I tried `N=60,70` and `80` and `s0` such that `yplus` value is about `30` but still got `Cd` about `0.0187` and the case kept on running even for `Time=5000`.

Then I reduced `yplus` value to about `20` but still it did not help. I don't what I'm missing. Here are my files:
[controlDict.txt](https://github.com/mdolab/dafoam/files/15087382/controlDict.txt)
[genWingMesh.py.txt](https://github.com/mdolab/dafoam/files/15087385/genWingMesh.py.txt)
[logPrimal.txt](https://github.com/mdolab/dafoam/files/15087386/logPrimal.txt)
[runScript_v2.py.txt](https://github.com/mdolab/dafoam/files/15087387/runScript_v2.py.txt)
[preProcessing.sh.txt](https://github.com/mdolab/dafoam/files/15087419/preProcessing.sh.txt)
![image](https://github.com/mdolab/dafoam/assets/118594705/c0cbe1bd-8f05-467b-998a-c2561ec0ccee)

**Note:** Solution was not converging in `time=1000` so I changed `endTime         5000;` in `controlDict`. Also, reduced tolerance to `"primalMinResTol": 1.0e-5,` but still no use.

**Reply by Mohsin-Bashir** [reply]
Is the value of density correct in runScript_v2.py? Should not it be more than 1?
![image](https://github.com/mdolab/dafoam/assets/118594705/241b0789-429a-49fa-bbdc-dd379a78cfe3)

**Reply by jCaudron** [reply]
Hi @Mohsin-Bashir, @friedenhe

It seems we're encountering similar issues. I've experimented with the parameters as suggested by @friedenhe, but unfortunately, I haven't had any success yet. Even with a y+ value around 3, the results remain unchanged.

Additionally, when I attempted optimization using "ipopt" and "slsqp," I encountered a stumbling block. On both occasions, I received the following error: "**EXIT: Converged to a point of local infeasibility. Problem may be infeasible.**"

@friedenhe, do you have any insights into what adjustments we might need to make to overcome this? Also, could you please clarify if there were any significant modifications made between your files and the ones provided in the tutorial? If so, sharing those details could be incredibly helpful for many DaFoam users.

Thank you for your assistance.

Best regards,
Julien

**Comment by hxl1310** [comment]
Thank you, I will try the grid settings you suggested.A large number of grids require high memory requirements for computers, and sometimes it is difficult to obtain accurate results
Thank you, I will try the grid settings you suggested

···生活···
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mdolab/dafoam"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年3月7日(星期四) 凌晨4:35
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mdolab/dafoam] The calculation result of case  Onera M6 wing is different from the tutorial result (Discussion #471)





 
We don't remember the exact N, s0 we used for the M6 tutorial figures. But I suggest you try to use N=60 to 80 and choose s0 such that the y+ is about 30.

**Comment by Mohsin-Bashir** [comment]
@hxl1310 @jCaudron have you been able to obtain the results mentioned on the website? If yes, kindly share the parameters used to obtain the results.
